feat(Quick Assistant):  Add the function to return from the Quick Assistant to the main window and continue the conversation

jwcrystal opened the pull request:
- Add IPC communication channels to implement conversation transfer between the Quick Assistant and the main window
- Add a "Return to Main Window" button at the bottom of the mini window and implement multi-language support
- Optimize the conversation data structure and storage logic to ensure data consistency

<!-- Template from https://github.com/kubevirt/kubevirt/blob/main/.github/PULL_REQUEST_TEMPLATE.md?-->
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/CherryHQ/cherry-studio/blob/main/CONTRIBUTING.md
-->

### What this PR does

Before this PR:

After this PR:

PR 解決快捷助手無法返回主視窗繼續對話的問題。
目前不太清楚存取 messages 有沒有更好寫法，這部分比較複雜

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->

Fixes #
延續 #6201 

### Why we need it and why it was done in this way

The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Breaking changes

<!-- optional -->

If this PR introduces breaking changes, please describe the changes and the impact on users.

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Documentation: A [user-guide update](https://docs.cherry-ai.com) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature.

### Release note

<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->

```release-note

```

<img width="300" alt="image" src="https://github.com/user-attachments/assets/bb40d09f-d4b0-42e6-ae24-173a8e5d0c50" />

jwcrystal committed:
feat: add quick assistant settings panel and management functionality

- Create QuickAssistantSettings component for UI
- Extend useAssistant hook with quick assistant controls
- Add settings button in ModelSettings page
- Implement temperature, context count, max tokens, and other parameters
- Connect settings to store via updateQuickAssistant action

Separate quick assistant preferences from default assistant settings for better customization.

jwcrystal committed:
refactor(QuickAssistantSettings): remove maxTokens and refine UI layout

- Removed maxTokens related state, logic, and UI elements
- Simplified settings page by eliminating unused configuration
- Adjusted layout for Slider and InputNumber for better usability
- Removed fixed width from Modal to enable responsive behavior

jwcrystal committed:
refactor(HomeWindow): optimize message building logic

- Removed redundant quickAssistant fetching logic
- Use `useQuickAssistant` hook directly for cleaner code
- Simplified message content concatenation method

jwcrystal committed:
Merge branch 'main' into feat/default-quick-assistant-panel

jwcrystal committed:
style(QuickAssistantSettings): Adjust spacing in settings page layout

Change the column width of sliders and input fields from 20/4 to 21/3 for a more reasonable layout
Also set the popup width to 800px to improve user experience

jwcrystal committed:
Merge branch 'main' into feat/default-quick-assistant-panel

jwcrystal committed:
feat(Quick Assistant): Add option to select assistant or model, and optimize Quick Assistant logic

- Added functionality to choose between using models or referencing other assistants
- Optimized model selection logic to automatically select based on settings
- Added relevant internationalization texts

jwcrystal committed:
fix(HomeWindow): Dynamically display input box placeholder text based on quick assistant states

jwcrystal committed:
refactor(QuickAssistant): remove the implement of the quick assistant feature and restructure related logic

- Remove code related to the quick assistant feature, including the useQuickAssistant hook, QuickAssistantSettings component, and associated store logic.
- Restructure the HomeWindow component to use default or specified assistants instead of the quick assistant functionality, simplifying the code structure.

jwcrystal committed:
feat(Quick Assistant): Add the function to return from the Quick Assistant to the main window and continue the conversation

- Add IPC communication channels to implement conversation transfer between the Quick Assistant and the main window
- Add a "Return to Main Window" button at the bottom of the mini window and implement multi-language support
- Optimize the conversation data structure and storage logic to ensure data consistency

jwcrystal committed:
refactor(home): Remove debug logs and clean code

- Remove electron-log debug logs from HomeWindow and HomePage components
- Clean up unused imports and commented-out code

jwcrystal committed:
fix(home): handle unprocessed errors when processing topics

jwcrystal commented:
ok，需要的話，這個我也可以改成通用版

0xfullex commented:
这个pr还包含上个未合并pr的内容？最好拆开一下

jwcrystal committed:
refactor(QuickAssistant): Remove custom default model for quick assistant and switch to default assistant

- Refactor quick assistant functionality, remove independent model settings, change to select via assistant ID
- Update multilingual translation text to match new features

jwcrystal commented:
> 这个pr还包含上个未合并pr的内容？最好拆开一下

我的錯，上傳順了...
目前有拆開的版本了，如果等另一個 pr 先合併的話，還需要先上傳嗎？ @0xfullex 
<img width="300" alt="image" src="https://github.com/user-attachments/assets/055c2bdf-4b5e-4ca2-9aef-63b3c5d1f77b" />

0xfullex commented:
> > 这个pr还包含上个未合并pr的内容？最好拆开一下
> 
> 我的錯，上傳順了... 目前有拆開的版本了，如果等另一個 pr 先合併的話，還需要先上傳嗎？ @0xfullex 

可以等另一个pr先完成

jwcrystal committed:
Merge branch 'feat/quick-assistant-topic-return-mian' into feat/default-quick-assistant-panel

jwcrystal committed:
refactor(window service): refactor the theme transfer logic from mini window to main window

- Isolate the window topic transfer logic
- Add new IPC channels and state management to handle theme transfer
- Introduce main window readiness state checking mechanism
- Optimize error handling and logging

jwcrystal commented:
目前已修改成通用函式 sendTopicToMainWindow，只要使用此函式皆可把訊息傳送到主視窗繼續話題討論，目前設計是每次都建立新話題討論。
使用 IpcChannel.App_Transfer_Topic_To_Main 去監聽

```tsx
 public async sendTopicToMainWindow(
    assistantId: string,
    topic: Topic, 
    sourceInfo: string = 'unknownSource'
  )
```

此次上傳的是有包含另一分支快捷助手使用助手模式的內容 #6201

jwcrystal committed:
refactor(QuickAssistant): Remove quick assistant-related states and simplify logic

- Remove unused quick assistant states and toggle functionality, simplifying related logic
- Update multilingual files to match the new default model and assistant labels

jwcrystal committed:
Merge branch 'feat/default-quick-assistant-panel' into feat/quick-assistant-topic-return-mian

jwcrystal committed:
refactor(i18n): Unify translation keys for input field placeholders

Unify the placeholder translation keys from `model_empty` and `assistant_empty` into empty across different scenarios, streamlining code logic

jwcrystal committed:
refactor(settings): simplify quick helper selection logic by directly using the preset helper

- Removed redundant helper filtering logic, directly using the preset helper as the quick helper

jwcrystal committed:
Merge branch 'feat/default-quick-assistant-panel' into feat/quick-assistant-topic-return-mian

jwcrystal committed:
Merge remote-tracking branch 'upstream/main' into feat/default-quick-assistant-panel

jwcrystal committed:
Merge branch 'feat/default-quick-assistant-panel' into feat/quick-assistant-topic-return-mian

jwcrystal committed:
refactor(assistants): remove unused upserTopic function
